test/clj/rems/api/test_applications.clj

```diff
@@ -741,15 +764,169 @@
+        (testing ", applicant can't fetch attachment"
+          (is (response-is-forbidden? (api-response :get (str "/api/applications/attachment/" attachment-id) nil
+                                                    api-key applicant-id))))))
+
+    (testing "handler approves with attachment"
+      (let [attachment-id (add-attachment handler-id (file "handler-approve.txt"))]
+        (is (number? attachment-id))
+        (is (= {:success true} (send-command handler-id
+                                             {:type :application.command/approve
+                                              :application-id application-id
+                                              :comment "see attachment"
+                                              :attachments [attachment-id]})))))
+
+    (testing "handler closes with two attachments"
+      (let [id1 (add-attachment handler-id (file "handler-close1.txt"))
+            id2 (add-attachment handler-id (file "handler-close2.txt"))]
+        (is (number? id1))
+        (is (number? id2))
+        (is (= {:success true} (send-command handler-id
+                                             {:type :application.command/close
+                                              :application-id application-id
+                                              :comment "see attachment"
+                                              :attachments [id1 id2]})))))
+
+    (testing "applicant can see the three new attachments"
+      (let [app (get-application application-id applicant-id)
+            [close-event approve-event] (reverse (:application/events app))
+            [close-id1 close-id2] (:event/attachments close-event)
+            [approve-id] (:event/attachments approve-event)]
+        (is (= "application.event/closed" (:event/type close-event)))
+        (is (= "application.event/approved" (:event/type approve-event)))
+        (is (number? close-id1))
+        (is (number? close-id2))
+        (is (number? approve-id))
+        (assert-response-is-ok (api-response :get (str "/api/applications/attachment/" close-id1) nil
+                                             api-key handler-id))
+        (assert-response-is-ok (api-response :get (str "/api/applications/attachment/" close-id2) nil
+                                             api-key handler-id))
+        (assert-response-is-ok (api-response :get (str "/api/applications/attachment/" approve-id) nil
+                                             api-key handler-id))))
+
+    (testing ":application/attachments"
+      (testing "applicant"
+        (is (= ["handler-remark.txt"
+                "handler-approve.txt"
+                "handler-close1.txt"
+                "handler-close2.txt"]
+               (mapv :attachment/filename (:application/attachments (get-application application-id applicant-id))))))
+      (testing "handler"
+        (is (= ["handler-remark.txt"
+                "reviewer-review.txt"
+                "handler-private.txt"
```

Comment:
Maybe name the `handler-remark.txt` and `handler-private.txt` as `handler-public-remark.txt` and `handler-private-remark.txt` so it's obvious that both are from remarks.

Reply:
done